planner/core/preprocess.go

```diff
@@ -315,32 +312,36 @@ func (p *preprocessor) checkAutoIncrement(stmt *ast.CreateTableStmt) {
 			}
 		}
 		if hasAutoIncrement {
-			count++
-			autoIncrementCol = colDef
+			autoIncrementCols[colDef] = isKey
 		}
 	}
 
-	if count < 1 {
+	if len(autoIncrementCols) < 1 {
 		return
-	}
-	if !isKey {
-		isKey = isConstraintKeyTp(stmt.Constraints, autoIncrementCol)
-	}
-	autoIncrementMustBeKey := true
-	for _, opt := range stmt.Options {
-		if opt.Tp == ast.TableOptionEngine && strings.EqualFold(opt.StrValue, "MyISAM") {
-			autoIncrementMustBeKey = false
-		}
-	}
-	if (autoIncrementMustBeKey && !isKey) || count > 1 {
+	} else if len(autoIncrementCols) > 1 {
```

Comment:
```suggestion
	if len(autoIncrementCols) > 1 {
```
The `if` statement is returned in line320.

Reply:
refactored.